File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        seen = defaultdict(int) 

        for i in range(len(numbers)):
            diff = target - numbers[i]

            if diff in seen:
                return [seen[diff] + 1, i + 1]
            
            seen[numbers[i]] = i
        return []
        

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        //USING HASHMAP

        HashMap<Integer, Integer> seen = new HashMap<>();
        
        for(int i = 0 ; i < numbers.length ; i++)
        {
            int diff = target - numbers[i];

            if(seen.containsKey(diff))
            {
                return new int []{seen.get(diff) + 1 , i + 1};
            }
            seen.put(numbers[i] , i);
        }
        return new int[]{-1 , -1};

    }
}